builtin/logical/pki/path_config_ca.go

```diff
@@ -83,16 +83,27 @@ func (b *backend) pathCAIssuersWrite(ctx context.Context, req *logical.Request,
 		return logical.ErrorResponse("Error resolving issuer reference: " + err.Error()), nil
 	}
 
+	response := &logical.Response{
+		Data: map[string]interface{}{
+			"default": parsedIssuer,
+		},
+	}
+
+	entry, err := fetchIssuerById(ctx, req.Storage, parsedIssuer)
+	if err != nil {
+		return logical.ErrorResponse("Unable to fetch issuer: " + err.Error()), nil
+	}
+
+	if len(entry.KeyID) == 0 {
+		response.AddWarning("This issuer has no key associated with it. Some operations like issuing certificates and signing CRLs will be unavailable with the requested default issuer.")
```

Comment:
Since this is so detrimental to the legacy api functioning, would it be worth also logging this to the Vault's log to possibly have support identify this as well?

Reply:
The issue is this isn't the only place where the default issuer can become disassociated from a key. We'd have to check, e.g., delete issuers, importing an issuer (with a cert and no key as this'll take the default slot), and perhaps some other paths for logging. Maybe its better to check at time of use? But `fetchCA...` currently already has resolved the issuer to an ID prior to the call I think, so it'd be hard to detect if it was the default, unless we cache that on the backend (which might actually be good)?

Reply:
Ok, I've updated this for the import case, @kitography will need to add it to their delete keys case to see if the deleted key was for the default issuer.

Reply:
Fixed.